ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/helpcenter/searcharticlepage/SearchArticlePresenterTest.java

```diff
@@ -0,0 +1,297 @@
+package com.kayako.sdk.android.k5.helpcenter.searcharticlepage;
+
+import com.kayako.sdk.android.k5.common.adapter.BaseListItem;
+import com.kayako.sdk.android.k5.common.adapter.searchlist.SearchListItem;
+import com.kayako.sdk.helpcenter.search.SearchArticle;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+public class SearchArticlePresenterTest {
+    private SearchArticlePresenter searchArticlePresenter;
+
+    @Mock
+    private SearchArticleContract.View mView;
+
+    @Mock
+    private SearchArticleContract.Data mData;
+
+    @Mock
+    private SearchArticle searchArticle;
+
+    private final static String TEST_VALUE = "test";
+
+    private List<SearchArticle> searchArticleList;
+
+    @Before
+    public void setUp() throws Exception {
+        searchArticlePresenter = new SearchArticlePresenter(mView, mData);
+        searchArticle.setTitle(TEST_VALUE);
+        searchArticle.setCategoryName(TEST_VALUE);
+        searchArticle.setSectionName(TEST_VALUE);
+        searchArticleList = Collections.singletonList(searchArticle);
+    }
+
+    @Test
+    public void initPage() throws Exception {
```

Comment:
✅ Good to see that test methods respect Java lower-camel-case and that Exception is thrown instead of specific exceptions.

This way the code respects Java naming conventions for methods and will avoid the risk of becoming "brittle" because they must be changed whenever the method under test changes.

**No change suggested, looks good!**	  

<a id="5999bf40-52d5-11e8-9c23-69d6096f3efc" href="none" codereview-uuid-reserved-tag />